python-lib/sharepoint_client.py

```diff
@@ -186,28 +186,47 @@ def setup_sharepoint_online_url(self, login_details):
         )
 
     def get_folders(self, path):
-        response = self.session.get(self.get_folder_url(path) + "/Folders")
+        url = self.get_folder_url() + "/Folders/{}".format(
+            self.get_path_as_query_string(path)
+        )
+        response = self.session.get(url)
         self.assert_response_ok(response, calling_method="get_folders")
         return response.json()
 
     def get_files(self, path):
-        response = self.session.get(self.get_folder_url(path) + "/Files")
+        response = self.session.get(self.get_folder_url() + "/Files/{}".format(
+                self.get_path_as_query_string(path)
+            )
+        )
         self.assert_response_ok(response, calling_method="get_files")
         return response.json()
 
     def get_item_fields(self, path):
-        response = self.session.get(self.get_folder_url(path) + "/ListItemAllFields")
+        response = self.session.get(self.get_folder_url() + "/ListItemAllFields{}".format(
+            self.get_path_as_query_string(path)
+        ))
         self.assert_response_ok(response, calling_method="get_item_fields")
         return response.json()
 
     def get_start_upload_url(self, path, upload_id):
-        return self.get_file_url(path) + "/startupload(uploadId=guid'{}')".format(upload_id)
+        return self.get_file_url() + "/startupload(uploadId=guid'{}'){}".format(
+            upload_id,
+            self.get_path_as_query_string(path)
+        )
```

Comment:
Similar point to the one I made on `get_folders()`  - you could have a `build_file_url` method just taking the last URL part.

Reply:
👍 planning it for sc-236752